frontend/src/metabase-types/api/settings.ts

```diff
@@ -181,12 +188,17 @@ export type PasswordComplexity = {
 
 export type SessionCookieSameSite = "lax" | "strict" | "none";
 
-export interface SettingDefinition {
-  key: string;
+export interface SettingDefinition<
+  Key extends EnterpriseSettingKey = EnterpriseSettingKey,
+> {
+  key: Key;
   env_name?: string;
   is_env_setting: boolean;
-  value?: unknown;
-  default?: unknown;
+  value: EnterpriseSettings[Key] | null;
```

Comment:
fixes type errors here. there's a JS file (I think SettingInput) that expects these values

Reply:
Does EnterpriseSettingsKey also contains non-enterprise setting keys? Would this work with non-enterprise settings?

Reply:
Actually, it looks like we have a type called `SettingElement` that actually has all of the types for the settings in the admin section. I think it'll be better to reuse that one and extend types off that